hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/ha/BootstrapStandby.java

```diff
@@ -199,28 +202,36 @@ private int doRun() throws IOException {
       return ERR_CODE_FAILED_CONNECT;
     }
 
-    if (!checkLayoutVersion(nsInfo)) {
-      LOG.error("Layout version on remote node (" + nsInfo.getLayoutVersion()
-          + ") does not match " + "this node's layout version ("
-          + HdfsServerConstants.NAMENODE_LAYOUT_VERSION + ")");
+    if (!checkLayoutVersion(nsInfo, isRollingUpgrade)) {
+      String msg = isRollingUpgrade ? "Layout version on remote node in rolling"
```

Comment:
This is not very readable to be honest.
Let's add a clear if and use the logger {} if possible.

Reply:
Added an if-else with the logger to make it more readable.